src/Microsoft.SqlTools.ResourceProvider.Core/Discovery/ServerTypes.cs

```diff
@@ -0,0 +1,66 @@
+﻿//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.SqlTools.ResourceProvider.Core
+{
+    /// <summary>
+    /// List of built-in server types used in <see cref="ExportableAttribute" />. 
+    /// Defines a server grouping based on the type of server connection supported (SQL Server, Reporting Server, Analysis Server)
+    /// Additional server types can be defined as needed.
+    /// Note that the Connection Dialog UI may require server type to be set for some resource types such as<see cref="IServerDiscoveryProvider" />. 
+    /// In addition a UI section matching that category may be required, or else the provider will not be used by any UI part and never be called.
+    /// </summary>
+    public static class ServerTypes
```

Comment:
>ServerTypes [](start = 24, length = 11)

at some point it may be worth seeing if we can merge some of the database\server object model code with the similar code we have in the toolsservice project to avoid duplicate, but different, object models.